src/Adapter/Currency/CommandHandler/AddCurrencyHandler.php

```diff
@@ -57,6 +75,7 @@ public function handle(AddCurrencyCommand $command)
             $entity->iso_code = $command->getIsoCode()->getValue();
             $entity->active = $command->isEnabled();
             $entity->conversion_rate = $command->getExchangeRate()->getValue();
+            $entity->refreshLocalizedCurrencyData(Language::getLanguages(), $this->localeRepoCLDR);
```

Comment:
I know this is an adapter handler and we will still have some Legacy code (starting with ObjectModel), but we should limit their use as much as we can So maybe you could use `LanguageRepositoryInterface` instead, and add it a `getLanguages` method (this interface is new we won't be able to change it in later versions so we might as well do it while we can)